newIDE/app/src/InstructionOrExpression/EnumerateInstructions.js

```diff
@@ -507,41 +509,32 @@ export const filterInstructionsList = <
   list: Array<T>,
   { searchText }: InstructionFilteringOptions
 ): Array<T> => {
-  if (!searchText === '') {
+  if (!searchText) {
     return list;
   }
 
-  const lowercaseSearch = searchText.toLowerCase();
-
-  const matchCritera = (enumeratedInstructionOrExpressionMetadata: T) => {
-    const {
-      displayedName,
-      fullGroupName,
-    } = enumeratedInstructionOrExpressionMetadata;
-    return (
-      displayedName.toLowerCase().indexOf(lowercaseSearch) !== -1 ||
-      fullGroupName.toLowerCase().indexOf(lowercaseSearch) !== -1
-    );
-  };
+  const directMatches = [];
+  const fuzzyMatches = [];
 
-  const favorExactMatch = (list: Array<T>): Array<T> => {
-    if (!searchText) {
-      return list;
+  list.forEach(option => {
+    // favor direct matches first
+    const lowercaseSearch = searchText.toLowerCase();
+    if (
+      option.displayedName.includes(searchText) ||
```

Comment:
Consider transforming to lowercase the displayed name before comparing to search text (so that the match is exact BUT case insensitive). Otherwise we won't match things with capital letters in the names.